pkg/sshd/lock_test.go

```diff
@@ -0,0 +1,85 @@
+package sshd
+
+import (
+	"testing"
+	"time"
+)
+
+const (
+	callbackTimeout = 1 * time.Second
+)
+
+func TestSingleLock(t *testing.T) {
+	rl := NewInMemoryRepositoryLock()
+	key := "fakeid"
+	callbackCh := make(chan interface{})
+	go lockAndCallback(rl, key, callbackCh)
+	verifyCallbackHappens(t, callbackCh)
+}
+
+func TestSingleLockUnlock(t *testing.T) {
+	rl := NewInMemoryRepositoryLock()
+	key := "fakeid"
+	callbackCh := make(chan interface{})
+	go lockAndCallback(rl, key, callbackCh)
+	verifyCallbackHappens(t, callbackCh)
+	err := rl.Unlock(key, time.Duration(0))
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+}
+
+func TestInvalidUnlock(t *testing.T) {
+	rl := NewInMemoryRepositoryLock()
+	key := "fakeid"
+	err := rl.Unlock(key, time.Duration(0))
+	if err == nil {
+		t.Fatalf("expected error but returned nil", err)
```

Comment:
this fails `go vet`: `no formatting directive in Fatalf call` I think this can remove the `err` (since it's nil) and make this a `t.Fatal`